OSDOCS-12974: 4.15.42 RN (#86646)
openshift/openshift-docs: release_notes/ocp-4-15-release-notes.adoc

For any {product-title} release, always review the instructions on xref:../updating/updating_a_cluster/updating-cluster-web-console.adoc#updating-cluster-web-console[updating your cluster] properly.
====

//4.15.42
[id="ocp-4-15-42_{context}"]
=== RHSA-2024:11562 - {product-title} 4.15.42 bug fix and security update

Issued: 02 January 2025

{product-title} release 4.15.42, which includes security updates, is now available. The list of bug fixes that are included in this update is documented in the link:https://access.redhat.com/errata/RHSA-2024:11562[RHSA-2024:11562] advisory. The RPM packages that are included in this update are provided by the link:https://access.redhat.com/errata/RHBA-2024:11565[RHBA-2024:11565] advisory.

Space precluded documenting all of the container images for this release in the advisory.

You can view the container images in this release by running the following command:

[source,terminal]
----
$ oc adm release info 4.15.42 --pullspecs
----

[id="ocp-4-15-42-bug-fixes_{context}"]
==== Bug fixes

> * Previously, when the webhook token authenticator was enabled and had the authorization type set to `None`, the {product-title} web console would consistently crash. With this release, a bug fix ensures that this configuration does not cause the {product-title} web console to crash. (link:https://issues.redhat.com/browse/OCPBUGS-46482[*OCPBUGS-46482*])

Reviewer: For bugs that don't have any complexity to the fix (where you end up just restating the issue), you an always lean on "The issue is resolved with this release" or similar :)

> * Previously, when the webhook token authenticator was enabled and had the authorization type set to `None`, the {product-title} web console would consistently crash. With this release, a bug fix ensures that this configuration does not cause the {product-title} web console to crash. (link:https://issues.redhat.com/browse/OCPBUGS-46482[*OCPBUGS-46482*])

Reviewer: For bugs that don't have any complexity to the fix (where you end up just restating the issue), you an always lean on "The issue is resolved with this release" or similar :)

> * Previously, when you attempted to use {olm-first} to upgrade an Operator, the upgrade was blocked and an `error validating existing CRs against new CRD's schema` message was generated. An issue existed with {olm}, whereby {olm} erroneously identified incompatibility issues validating existing custom resources (CRs) against the new Operator version's custom resource definitions (CRDs). With this release, the validation is corrected so that Operator upgrades are no longer blocked. (link:https://issues.redhat.com/browse/OCPBUGS-46479[*OCPBUGS-46479*])

Reviewer: `{olm-first}`  and `{olm}` don't seem to be defined

> * Previously, the images for custom OS layering were not present when the OS was on {rhcos-short} 4.15, preventing some customers from upgrading from {rhcos-short} 4.15 to {rhcos-short} 4.16. This release adds ACR and GCR image credential provider RPMs to {rhcos-short} 4.15. (link:https://issues.redhat.com/browse/OCPBUGS-46063[*OCPBUGS-46063*])

Reviewer: `{rhcos-short}` is also undefined. Do ACR and GCR need to be written out?

Author: done :D

> * Previously, it was impossible to configure your Amazon Web Services DHCP option set with a custom domain name containing a trailing period, as trailing periods were not allowed in a Kubernetes object name. With this release, trailing periods are allowed in a domain name in a DHCP option set. (link:https://issues.redhat.com/browse/OCPBUGS-46034[*OCPBUGS-46034*])

Reviewer: I'm not 100% sure, but is "trailing period" something like `domainname.`? Might rephrase as:
```suggestion
* Previously, you could not configure your Amazon Web Services DHCP option set with a custom domain name containing a period (`.`) as the final character, as trailing periods were not allowed in a Kubernetes object name. With this release, trailing periods are allowed in a domain name in a DHCP option set. (link:https://issues.redhat.com/browse/OCPBUGS-46034[*OCPBUGS-46034*])
```

* Previously, when `openshift-sdn` pods were deployed during the {product-title} upgrading process, the Open vSwitch (OVS) storage table was cleared. This issue occurred on {product-title} {product-version}.19 and later versions. Ports for existing pods had to be re-created and this disrupted numerous services. With this release, a fix ensures that the OVS tables do not get cleared and pods do not get disconnected during a cluster upgrade operation. (link:https://issues.redhat.com/browse/OCPBUGS-45955[*OCPBUGS-45955*])

> * Previously, you could not remove a finally pipeline task from the *edit Pipeline* form if you created a pipeline with only one finally task. With this release, you can remove the finally task from the *edit Pipeline* form and the issue is resolved. (link:https://issues.redhat.com/browse/OCPBUGS-45950[*OCPBUGS-45950*])

Reviewer: Suggest putting "finally" in backticks (like [here](https://tekton.dev/docs/pipelines/pipelines/)) because it looks like a typo otherwise :sweat_smile: 
```suggestion
* Previously, you could not remove a `finally` pipeline task from the *edit Pipeline* form if you created a pipeline with only one `finally` task. With this release, you can remove the `finally` task from the *edit Pipeline* form and the issue is resolved. (link:https://issues.redhat.com/browse/OCPBUGS-45950[*OCPBUGS-45950*])
```

Author: true! great catch!

> * Previously, the `aws-sdk-go-v2` software development kit (SDK) failed to authenticate an `AssumeRoleWithWebIdentity` API operation on an {aws-first} {sts-first} cluster. With this release, `pod-identity-webhook` now includes a default region, and this issue no longer persists. (link:https://issues.redhat.com/browse/OCPBUGS-45940[*OCPBUGS-45940*])

Reviewer: ```suggestion
* Previously, the `aws-sdk-go-v2` software development kit (SDK) failed to authenticate an `AssumeRoleWithWebIdentity` API operation on an {aws-short} {sts-first} cluster. With this release, the pod identity webhook now includes a default region, and this issue no longer persists. (link:https://issues.redhat.com/browse/OCPBUGS-45940[*OCPBUGS-45940*])
```

* Previously, the installation program populated the `network.devices`, `template` and `workspace` fields in the `spec.template.spec.providerSpec.value` section of the {vmw-full} control plane machine set custom resource (CR).
These fields should be set in the {vmw-short} failure domain, and the installation program populating them caused unintended behaviors.
Updating these fields did not trigger an update to the control plane machines, and these fields were cleared when the control plane machine set was deleted.
With this release, the installation program is updated to no longer populate values that are included in the failure domain configuration.
If these values are not defined in a failure domain configuration, for instance on a cluster that is updated to {product-title} {product-version} from an earlier version, the values defined by the installation program are used.
(link:https://issues.redhat.com/browse/OCPBUGS-45839[*OCPBUGS-45839*], link:https://issues.redhat.com/browse/OCPBUGS-37064[*OCPBUGS-37064*])

> * Previously, the ClusterTasks were listed on the *Pipelines builder* and *ClusterTask list* pages in the *Tasks* navigation menu. With this release, the ClusterTasks are deprecated from Pipelines 1.17 and the ClusterTask dependency is removed from static plug-in. On the Pipelines builder page, you will only see the present task in the Namespace and Community tasks. (link:https://issues.redhat.com/browse/OCPBUGS-45248[*OCPBUGS-45248*])

Reviewer: Based on use in https://docs.openshift.com/pipelines/1.17/create/remote-pipelines-tasks-resolvers.html
```suggestion
* Previously, the ClusterTasks were listed on the *Pipelines builder* and *ClusterTask list* pages in the *Tasks* navigation menu. With this release, the `ClusterTask` functionality is deprecated from Pipelines 1.17 and the `ClusterTask` dependency is removed from static plug-in. On the Pipelines builder page, you will only see the present task in the Namespace and Community tasks. (link:https://issues.redhat.com/browse/OCPBUGS-45248[*OCPBUGS-45248*])
```
